lesson_15/hw_15.py

```diff
@@ -12,4 +12,51 @@
     Протилежні кути ромба завжди рівні, тому при заданому значенні кут_а, значення
     кут_б обчислюється автоматично.
     Для встановлення значень атрибутів використовуйте метод __setattr__.
-"""
\ No newline at end of file
+"""
+
+
+class Rhombus:
+    def __init__(self, side_a, angle_a):
+        self.side_a = side_a
+        self.angle_a = angle_a
+
+    def __setattr__(self, name, value):
+        if isinstance(value, (int, float)):
+
+            if name == "side_a":
+
+                if value <= 0:
+                    raise ValueError("Length must be positive")
+
+            if name == "angle_a":
+
+                if not (0 < value < 180):
+                    raise ValueError("Angle should be between 0 and 180 degree")
+
+                super().__setattr__("angle_b", 180 - value)
+
+            if name == "angle_b":
+
+                if not (0 < value < 180):
+                    raise ValueError("Angle should be between 0 and 180 degree")
+
+                super().__setattr__("angle_a", 180 - value)
+
+            super().__setattr__(name, value)
```

Comment:
```suggestion
            super().__setattr__(name, value)
```
зверніть увагу - це виконується незалежно ввід усіх попередніх іф.
Подумайте, до яких проблем це може призвести і як це краще зробити